prow/sidecar/run.go

```diff
@@ -121,13 +122,29 @@ func (o Options) Run() error {
 		aborted = returnCode == 130
 	}
 
-	return o.doUpload(spec, passed, aborted)
+	metadata := map[string]string{}
+	metadataRaw := o.GcsOptions.GetNamedArtifactsContent("metadata.json")
+	if len(metadataRaw) > 0 {
+		if err := json.Unmarshal(metadataRaw, &metadata); err != nil {
+			logrus.WithError(err).Warn("Failed to unmarshal metadata.json")
+		}
+	}
+
+	return o.doUpload(spec, passed, aborted, metadata)
```

Comment:
Before we call `doUpload()` we should populate some of the common metadata fields if they are not already set. That way jobs will only have to specify metadata.json if they have custom metadata fields or if they want to overwrite a metadata field.

Reply:
so metadata.json is generated by kubetest, prow doesn't really have any knowledge about that :-\

I'll add the main repo SHA in finished.json if that's presented.

Reply:
The git refs that are checked out was the main thing I was thinking of. If we find that `bootstrap.py` generates other metadata fields that we should include in the pod utilities we can add those later.

Reply:
bootstrap also try to add a test-infra SHA, but since we don't checkout test-infra anymore, we can probably drop that.